helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -893,42 +940,176 @@ private RollingUpdateOp rollingUpdateDeploy(final ZooKeeperClient client,
     }
   }
 
+  private RollingUpdateOp rollingUpdateAwaitStopped(final ZooKeeperClient client,
```

Comment:
Agian, I think the name should be `...Undeployed` rather than `...Stopped`, since we want the job undeployed and not stopped.

Also I think this entire method could be simplified a lot:

```
  private RollingUpdateOp rollingUpdateAwaitUndeployed(final ZooKeeperClient client,
                                                       final RollingUpdateOpFactory opFactory,
                                                       final DeploymentGroup deploymentGroup,
                                                       final String host) {
    // Yield until the job is undeployed on the given host, or the rollout times out.
    // Once the job is fully undeployed "task status" path for it, i.e.
    // /status/hosts/<host>/job/<job-id>
    final TaskStatus taskStatus = getTaskStatus(client, host, deploymentGroup.getJobId());
    if (taskStatus == null) {
      // The job is undeployed!
      return opFactory.nextTask();
    } else if (isRolloutTimedOut(client, deploymentGroup)) {
        return opFactory.error("timed out while retrieving job status", host,
                               RollingUpdateError.TIMED_OUT_WAITING_FOR_JOB_TO_UNDEPLOY);
    } else {
      return opFactory.yield();
    }
  }
```

(The `DeploymentGroupTest` system tests pass with this change)